### PR TITLE
call setEvn before set quota

### LINKED ISSUE #878: [BUG] SetQuota failed when I  set env SENTINEL_PASSWORD
**What happened**:
```
I0227 08:59:17.477030       7 juicefs.go:812] SetQuota cmd: /usr/local/bin/juicefs quota set ${metaurl} --path pvc-0236ab44-ef86-4ddd-97b9-34a695dcf743 --capacity 50
E0227 08:59:17.578058       7 node.go:169] set quota: 2024/02/27 08:59:17.572378 juicefs[70] <INFO>: Meta address: redis://:****@master,192.168.212.33,192.168.212.34,192.168.212.35:36379/1 [interface.go:497]
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
2024/02/27 08:59:17.574588 juicefs[70] <WARNING>: parse info: redis: all sentinels specified in configuration are unreachable [redis.go:3575]
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
redis: 2024/02/27 08:59:17 sentinel.go:537: sentinel: GetMasterAddrByName master="master" failed: NOAUTH HELLO must be called with the client already authenticated, otherwise the HELLO <proto> AUTH <user> <pass> option can be used to authenticate the client and select the RESP protocol version at the same time
2024/02/27 08:59:17.575713 juicefs[70] <FATAL>: Load setting: redis: all sentinels specified in configuration are unreachable [quota.go:127]
```

**How to reproduce it (as minimally and precisely as possible)**:
```
apiVersion: v1
kind: Secret
metadata:
  name: juicefs-sc-secret
data:
  access-key: xxx
  bucket: *****
  envs: {"SENTINEL_PASSWORD":"********"}
  metaurl: redis://:********@master,192.168.212.33,192.168.212.34,192.168.212.35:36379/1
  name: ******
  secret-key: *****
  storage: ****
type: Opaque
```

set SENTINEL_PASSWORD environment variable

**Environment**:
- JuiceFS CSI Driver version (which image tag did your CSI Driver use):
- **v0.23.3**
- Kubernetes version (e.g. `kubectl version`):
- **v1.18**
- Object storage (cloud provider and region):
- **minio**
- Metadata engine info (version, cloud provider managed or self maintained):
- **redis sentinel**

### PR DESCRIPTION
**Pull request by 1iza**
issues #878

### CODE REVIEW

**Review by zxh326 — APPROVED** [review]
LGTM, thx

/cc @timfeirg

File: pkg/juicefs/juicefs.go [hunk 1]
```diff
@@ -812,15 +812,24 @@ func (j *juicefs) SetQuota(ctx context.Context, secrets map[string]string, jfsSe
 	klog.Infof("SetQuota cmd: %s", strings.Join(cmdArgs, " "))
 	cmdCtx, cmdCancel := context.WithTimeout(ctx, 2*defaultCheckTimeout)
 	defer cmdCancel()
-
+	envs := syscall.Environ()
+	for key, val := range jfsSetting.Envs {
+		envs = append(envs, fmt.Sprintf("%s=%s", security.EscapeBashStr(key), security.EscapeBashStr(val)))
+	}
+	if secrets["storage"] == "ceph" || secrets["storage"] == "gs" {
+		envs = append(envs, "JFS_NO_CHECK_OBJECT_STORAGE=1")
```
Comment (timfeirg) [inline pkg/juicefs/juicefs.go]: this variable only affects the mount subcommand, omit this

Comment (1iza) [reply pkg/juicefs/juicefs.go]: done

**Comment by timfeirg** [comment]
this seems like a reasonable change, I'll take over this pr and move to upstream, all of our e2e tests run exclusively in the upstream repo CI.

closing in favor of https://github.com/juicedata/juicefs-csi-driver/pull/885

**Pull request closed without merging**